File: frontend/src/app/models/incidencia.ts
export type EstadoIncidencia = 'PENDIENTE' | 'EN_PROCESO' | 'RESUELTA';

export interface Incidencia {
  id: number;
  titulo: string;
  descripcion: string;
  estado: EstadoIncidencia;
}

export type IncidenciaRequest = Omit<Incidencia, 'id'>;

File: backend/src/main/java/com/PC2/backend/entity/Incidencia.java
package com.PC2.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "incidencias")
public class Incidencia {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, length = 150)
	private String titulo;

	@Column(nullable = false, length = 500)
	private String descripcion;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false, length = 30)
	private EstadoIncidencia estado = EstadoIncidencia.PENDIENTE;
}
